base/compiler/utilities.jl

```diff
@@ -114,12 +114,29 @@ function get_staged(li::MethodInstance)
     end
 end
 
+function has_opaque_closure(c::CodeInfo)
+    for i = 1:length(c.code)
+        stmt = c.code[i]
+        (isa(stmt, Expr) && stmt.head === :new_opaque_closure) && return true
+    end
+    return false
+end
+
 function retrieve_code_info(linfo::MethodInstance)
     m = linfo.def::Method
     c = nothing
     if isdefined(m, :generator)
-        # user code might throw errors – ignore them
-        c = get_staged(linfo)
+        if isdefined(linfo, :uninferred)
+            c = copy(linfo.uninferred::CodeInfo)
+        else
+            # user code might throw errors – ignore them
+            c = get_staged(linfo)
+            # For opaque closures, cache the generated code info to make sure
```

Comment:
Since this is a correctness issue, and there's many places it gets called, `get_staged` must be handling this itself

Reply:
I figured it was only a correctness issue if it escaped into inference, which should always happen here. Doing the copy in `C` is kinda annoying, but I guess I'll just move it there.

Reply:
Actually, I guess I can just move this into the `get_staged` julia-side function, which avoids duplicating the copy logic on the C side. The other callers from C just use it to generate code directly, which doesn't have the correctness issues.